packages/core/src/model.ts

```diff
@@ -0,0 +1,35 @@
+import { assign } from './actions';
+import { AssignAction, Assigner, EventObject, PropertyAssigner } from './types';
+import { mapValues } from './utils';
+
+export function createModel<TContext, TEvent extends EventObject>(
+  initialState: TContext
+): {
+  context: TContext;
+  withUpdaters: <
+    T extends {
+      [key: string]:
+        | Assigner<TContext, TEvent>
+        | PropertyAssigner<TContext, TEvent>;
+    }
+  >(
+    assigners: T
+  ) => {
+    context: TContext;
+    actions: {
+      [K in keyof T]: AssignAction<TContext, TEvent>;
+    };
+  };
+} {
+  return {
+    context: initialState,
```

Comment:
this is only ever be `initialContext`/`initialState`, so maybe it would be good to include that in the property name as right now one could think that the while model is impure and that it can be mutated/updates, whereas it's just a "base" for contexts of machines

Reply:
Renamed to `initialContext`